src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -625,23 +631,53 @@ static inline int MPIDI_NM_mpi_init_hook(int rank,
         sprintf(keyS, "OFI-%d", rank);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Put(MPIDI_Global.kvsname, keyS, val), pmi);
         MPIDI_OFI_PMI_CALL_POP(PMI_KVS_Commit(MPIDI_Global.kvsname), pmi);
-        MPIDI_OFI_PMI_CALL_POP(PMI_Barrier(), pmi);
+
+        local_procs = MPL_malloc(size * sizeof(int));
+        local_ranks = MPL_malloc(size * sizeof(int));
+
+        for (i = 0; i < size; i++) {
+            if (MPIDI_CH4_rank_is_local(i, MPIR_Process.comm_world)) {
+                if (i == rank) {
+                    local_rank = num_local;
+                }
+
+                local_procs[num_local] = i;
+                local_ranks[i] = num_local;
+                num_local++;
+            }
+        }
```

Comment:
local_ranks is completely unused and should be removed.

local_procs is not needed.  We only need the local root and the local rank.